src/coreclr/jit/importer.cpp

```diff
@@ -4449,7 +4449,15 @@ GenTree* Compiler::impImportStaticFieldAccess(CORINFO_RESOLVED_TOKEN* pResolvedT
                     callFlags |= GTF_CALL_HOISTABLE;
                 }
 
-                op1 = gtNewHelperCallNode(CORINFO_HELP_READYTORUN_STATIC_BASE, TYP_BYREF);
+                op1 = gtNewHelperCallNode(pFieldInfo->helper, TYP_BYREF);
+                if (pResolvedToken->hClass == info.compClassHnd)
+                {
+                    if (pFieldInfo->helper == CORINFO_HELP_READYTORUN_GCSTATIC_BASE ||
+                        pFieldInfo->helper == CORINFO_HELP_READYTORUN_NONGCSTATIC_BASE)
+                    {
+                        m_preferredInitCctor = pFieldInfo->helper;
```

Comment:
Does it make any difference if we only do this once? (the idea being that the earlier we see this, the more likely it is that we can CSE it)